### PR TITLE
External Integrations

### PR DESCRIPTION
**Pull request by AndyTWF**
- Add hidden window to allow detection that UKCP is running.
- Add hidden window message for stand assignment.

### CODE REVIEW

File: src/integration/HiddenWindow.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                case WM_CREATE: {
+                    handler = reinterpret_cast<ExternalMessageEventHandler*>(
+                        reinterpret_cast<CREATESTRUCT*>(lParam)->lpCreateParams
+                    );
+                    return TRUE;
+                }
+                case WM_COPYDATA: {
+                    COPYDATASTRUCT* data = reinterpret_cast<COPYDATASTRUCT*>(lParam);
+
+                    if (data != nullptr && data->dwData == 811 && data->lpData != nullptr && handler != nullptr) {
```
Comment (AxonC) [inline src/integration/HiddenWindow.cpp:21]: 811 seems a little magic to me 🗡️

**Review by AxonC — CHANGES_REQUESTED** [review]

File: src/integration/HiddenWindow.cpp (hunk 1 above)
Comment (AndyTWF) [reply src/integration/HiddenWindow.cpp:21]: You just have to pick a number so I came up with a random one :D This is the one we've sent to the vstrips team, so probably best we don't change it if we can help it :)

Comment (AxonC) [reply src/integration/HiddenWindow.cpp:21]: Can I have a comment or constant to that effect?

Comment (AndyTWF) [reply src/integration/HiddenWindow.cpp]: Done!

**Review by AxonC — APPROVED** [review]

**Comment by Heath123** [comment]
This showed up on the globe on the GitHub main page by the way

**Pull request merged**